tests/dist/python/run_dist_objects.py

```diff
@@ -48,6 +51,239 @@ def run_server(
     g.start()
 
 
+##########################################
+############### DistGraph ###############
+##########################################
+
+def node_split_test(g, force_even, ntype='_N'):
+    all_nodes_mask = F.arange(0, g.number_of_nodes(ntype))
+
+    gpb = g.get_partition_book()
+    selected_nodes_dist_tensor = dgl.distributed.DistTensor([g.number_of_nodes(ntype)], F.int32, init_func=rand_init)
+    selected_nodes = F.asnumpy(selected_nodes_dist_tensor[all_nodes_mask]) == 1
+    g.barrier()
+
+    nodes = node_split(selected_nodes, gpb, ntype=ntype, force_even=force_even)
+    selected_nodes_dist_tensor[nodes] = F.astype(F.zeros_like(nodes), selected_nodes_dist_tensor.dtype)
+    g.barrier()
+
+    assert F.sum(selected_nodes_dist_tensor[all_nodes_mask], 0) == 0
+
+    g.barrier()
+
+def edge_split_test(g, force_even, etype='_E'):
+    all_edges_mask = F.arange(0, g.number_of_edges(etype))
+
+    selected_edges_dist_tensor = dgl.distributed.DistTensor([g.number_of_edges(etype)], F.int32, init_func=rand_init)
+    selected_edges = F.asnumpy(selected_edges_dist_tensor[all_edges_mask]) == 1
+    g.barrier()
+
+    edges = edge_split(selected_edges, gpb, etype=etype, force_even=force_even)
+
+    selected_edges_dist_tensor[edges] = F.astype(F.zeros_like(edges), selected_edges_dist_tensor.dtype)
+    g.barrier()
+    assert F.sum(selected_edges_dist_tensor[all_edges_mask], 0) == 0
+
+    g.barrier()
+
+def dist_graph_test(g):
+    gpb_path = graph_path + '/{}.json'.format(graph_name)
+    with open(gpb_path) as conf_f:
+        part_metadata = json.load(conf_f)
+    assert 'num_nodes' in part_metadata
+    assert 'num_edges' in part_metadata
+    num_nodes = part_metadata['num_nodes']
+    num_edges = part_metadata['num_edges']
+
+    assert g.number_of_nodes() == num_nodes
+    assert g.number_of_edges() == num_edges
+
+    assert g.node_attr_schemes()['feat'].dtype == F.int64
+    assert g.node_attr_schemes()['feat'].shape == (1,)
+
+    node_split_test(g, force_even=False)
+    node_split_test(g, force_even=True)
+
+    edge_split_test(g, force_even=False)
+    edge_split_test(g, force_even=True)
+
+def dist_graph_hetero_test(g):
```

Comment:
could we unify and merge test on homogeneous and heterogeneous graph into single one test function as much code could be shared? Simply speaking, we can access homograph in the same way as hetero.